lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -82,80 +160,88 @@ def is_active(self) -> bool:
             logger.error("Error while checking Vault health status: %s", e)
             return False
 
-    def is_api_available(self) -> bool:
-        """Return whether Vault is available."""
-        try:
-            self._client.sys.read_health_status(standby_ok=True)
-            return True
-        except (VaultError, RequestException) as e:
-            logger.error("Error while checking Vault health status: %s", e)
-            return False
+    def enable_audit_device(self, device_type: AuditDeviceType, path: str) -> None:
+        """Enable a new audit device at the supplied path.
 
-    def unseal(self, unseal_keys: List[str]) -> None:
-        """Unseal Vault."""
-        for unseal_key in unseal_keys:
-            self._client.sys.submit_unseal_key(unseal_key)
-        logger.info("Vault is unsealed")
-
-    def set_token(self, token: str) -> None:
-        """Set the Vault token for authentication."""
-        self._client.token = token
+        Args:
+            device_type: One of three available device types
+            path: The path that will receive audit logs
+        """
+        try:
+            self._client.sys.enable_audit_device(
+                device_type=device_type.value,
+                options={"file_path": path},
+            )
+            logger.info("Enabled audit device %s for path %s", device_type.value, path)
+        except InvalidRequest:
+            logger.info("Audit device already enabled.")
 
-    def approle_login(self, role_id: str, secret_id: str) -> None:
-        """Authenticate to Vault using AppRole."""
-        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+    def enable_approle_auth_method(self) -> None:
+        """Enable approle auth method."""
+        try:
+            self._client.sys.enable_auth_method("approle")
+            logger.info("Enabled approle auth method.")
+        except InvalidRequest:
+            logger.info("Approle already enabled.")
```

Comment:
Good question. The exceptions in `hvac` are mostly a mapping of the [HTTP status codes](https://developer.hashicorp.com/vault/api-docs#400) that vault returns. The confidence comes from knowing that we filter the wrong parameters before we ever make the call to `hvac` in the first place. If a regression happens where this exception is being raised because of a different reason, the solution would be to filter that out in python before we make the call to `hvac` rather than having to read and filter through the InvalidRequest exception.

The assumptions here could change in the future, but we can handle that in the future when this method does do what you say it might.

Reply:
It seems to me that we want to rely on these exceptions to validate that approle or audit devices were actually enabled and that can lead to false positives. This is my main concern.

Reply:
I thought about this a bit more, and decided that we may as well do it now. changed.

Reply:
Wouldn't it be a more robust approach to actually check if these are enabled instead of relying on the exception?